src/main/java/com/auction/auction/bid/service/AuctionService.java

```diff
@@ -121,5 +107,25 @@ public List<Auction> getAllAuctions() {
         return auctionRepository.findAll();
     }
 
+    @Override
+    public Auction saveAuction(Auction auction) {
+        if (auction.getStatus() == AuctionStatus.IN_PROCESS || auction.getStatus() == AuctionStatus.COMPLETED) {
```

Comment:
In that case i would reccomend you to use additional value in AuctionStatus. Like:
 COMPLETED(isClosable),
 IN_PROCESS(isClosable)...
And "isClosable" is boolean value. 
So your implementation will be : if (auction.getStatus().isClosable) {}
In such way if tomorrow you will add one more status to auction you will have to set isClosable to it, and you should not update you current code